internal/storage/caching/fast_stat_bucket.go

```diff
@@ -83,6 +85,36 @@ func (b *fastStatBucket) insertMultiple(objs []*gcs.Object) {
 	}
 }
 
+// InsertHierarchicalListing saves the objects in cache excluding zero byte objects corresponding to folders
+// by iterating objects present in listing and saves prefixes as folders all prefixes are folders in hns by
+// iterating collapsedRuns of listing.
+func (b *fastStatBucket) insertHierarchicalListing(listing *gcs.Listing) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	expiration := b.clock.Now().Add(b.ttl)
+
+	for _, o := range listing.Objects {
+		if !strings.HasSuffix(o.Name, "/") {
+			m := storageutil.ConvertObjToMinObject(o)
+			b.cache.Insert(m, expiration)
+		}
+	}
+
+	for _, p := range listing.CollapsedRuns {
+		if !strings.HasSuffix(p, "/") {
+			// log the error for incorrect prefix but don't fail the operation
+			logger.Errorf("error in prefix name: %s", p)
```

Comment:
Should this be a warning instead of an error? Asking this because this is a GCS-caused issue, and it does not affect user in any way.

Reply:
Anyway this error/warning must have been also logged at the `bucketHandle` level, so it's just piling on error message after error message for the same issue.

Reply:
Anyway this error/warning must have been also logged at the bucketHandle level, so it's just piling on error message after error message for the same issue. > since this error is gulped here and not returned it will not be logged at bucket handle level, 
added error level here as this is an error only, should be logged at log level info also